stdlib/public/core/StringUTF8.swift

```diff
@@ -545,33 +614,51 @@ extension String.UTF8View.Iterator : IteratorProtocol {
   }
 }
 
+// Used to calculate a running count. For non-BMP scalars, it's important if the
+// prior code unit was a leading surrogate (validity).
+internal func _utf8Count(_ utf16CU: UInt16, prev: UInt16) -> Int {
+  switch utf16CU {
+  case 0..<0x80: return 1
+  case 0x80..<0x800: return 2
+  case 0x800..<0xDC00: return 3
+  case 0xDC00..<0xE000: return UTF16.isLeadSurrogate(prev) ? 1 : 3
+  default: return 3
+  }
+}
+
 extension String.UTF8View {
-  @inlinable // FIXME(sil-serialize-all)
+  @usableFromInline
```

Comment:
I think this is not used from inlinable code anymore, so `@usableFromInline` may be removed.

Reply:
Ah, good point. I had done a purge of many of these in another PR and I'll try another one. Also, I think in general we could use a diagnostic pass letting us know.

https://bugs.swift.org/browse/SR-7687